package/kedro_viz/models/utils.py

```diff
@@ -39,3 +45,45 @@ def get_dataset_type(dataset: "AbstractDataset") -> str:
     abbreviated_module_name = ".".join(dataset.__class__.__module__.split(".")[-2:])
     class_name = f"{dataset.__class__.__qualname__}"
     return f"{abbreviated_module_name}.{class_name}"
+
+
+def get_file_size(file_path: Union[str, None]) -> Union[int, None]:
+    """Get the dataset file size using fsspec. If the file_path is a directory,
+    get the latest file created (this corresponds to the latest run)
+
```

Comment:
Why can't we calculate this from the hook? It should be much simpler to do this if you have the `Dataset` object. That way you don't need to re-create all the `fsspec` logic.

Reply:
I tried doing that using _filepath in after_context_created hook. But since AbstractDataset does not have an api supporting that, you mentioned it to be fragile and so thought it would be a good idea to move it here